### PR TITLE
[CASSANDRA-656] Consume zones as racks for Cassandra

### PR DESCRIPTION
**Pull request by gabrielhartmann**

### CODE REVIEW

File: frameworks/cassandra/universe/config.json [hunk 1]
```diff
@@ -35,10 +35,10 @@
           "type": "string",
           "default": "datacenter1"
         },
-        "rack": {
-          "description": "The name of the rack this cluster is running on",
-          "type": "string",
-          "default": "rack1"
+        "detect_zones": {
+          "description": "Whether to use Zones when available for Cassandra racks.  This value cannot be changed after initial deployment",
```
Comment (gabrielhartmann) [inline frameworks/cassandra/universe/config.json]: @sascala This description is very poorly worded.  'lil help please.

File: frameworks/cassandra/tests/test_racks.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        yield # let the test session execute
+    finally:
+        sdk_install.uninstall(config.PACKAGE_NAME, config.SERVICE_NAME)
+
+@pytest.mark.sanity
+def test_rack():
+    sdk_install.install(config.PACKAGE_NAME, config.SERVICE_NAME, 3)
+
+    # dcos task exec node-0-server bash -c 'JAVA_HOME=jre1.8.0_144 apache-cassandra-3.0.14/bin/nodetool status'
+    raw_status = sdk_tasks.task_exec('node-0-server', "bash -c 'JAVA_HOME=jre1.8.0_144 apache-cassandra-3.0.14/bin/nodetool status'")
```
Comment (triclambert) [inline frameworks/cassandra/tests/test_racks.py]: These hard coded values are likely to get stale quickly.

File: frameworks/cassandra/universe/marathon.json.mustache [hunk 3]
```diff
@@ -153,7 +154,6 @@
     "CASSANDRA_DOCKER_IMAGE": "{{resource.assets.container.docker.cassandra}}",
     "JAVA_URI": "{{resource.assets.uris.jre-tar-gz}}",
     "TASKCFG_ALL_CASSANDRA_LOCATION_DATA_CENTER": "{{service.data_center}}",
-    "TASKCFG_ALL_CASSANDRA_LOCATION_RACK": "{{service.rack}}",
```
Comment (triclambert) [inline frameworks/cassandra/universe/marathon.json.mustache]: Are there any downsides to removing this config value?  Just checking.

Comment (gabrielhartmann) [reply frameworks/cassandra/universe/marathon.json.mustache]: Having an end-user set the "rack" string was useless from the beginning.  Everything was always on the same rack named the same thing.

File: frameworks/cassandra/tests/test_racks.py (hunk 2 above)
Comment (gabrielhartmann) [reply frameworks/cassandra/tests/test_racks.py]: Affirmative

Comment (benclarkwood) [reply frameworks/cassandra/tests/test_racks.py]: `JAVA_HOME=$(ls -d $MESOS_SANDBOX/jre*/)`

Comment (benclarkwood) [reply frameworks/cassandra/tests/test_racks.py]: Can you put the CASSANDRA_VERSION into the task env?

Comment (benclarkwood) [reply frameworks/cassandra/tests/test_racks.py]: Or something like: `CASSANDRA_VERSION = sdk_marathon.get_config(foldered_name)['env']['CASSANDRA_VERSION']`

Comment (nickbp) [reply frameworks/cassandra/tests/test_racks.py]: You can just use asterisks. Did it in this PR: https://github.com/mesosphere/dcos-commons/pull/1809/files#diff-fa78f986c873acfad32eed0038c64638R149

**Comment by gabrielhartmann** [comment]
@nickbp @benclarkwood:  Fixed the nodetool command.  Added updated `bootstrap` to fix mustache rendering.

File: frameworks/cassandra/universe/marathon.json.mustache (hunk 3 above)
Comment (gabrielhartmann) [reply frameworks/cassandra/universe/marathon.json.mustache]: @triclambert spoke with the relevant authorities.  Removing this is fine.

**Pull request merged**